proxy/src/auth/backend.rs

```diff
@@ -128,22 +129,51 @@ pub struct AuthSuccess<T> {
     pub value: T,
 }
 
+impl<T> AuthSuccess<T> {
+    /// Very similar to [`std::option::Option::map`].
+    /// Maps [`AuthSuccess<T>`] to [`AuthSuccess<R>`] by applying
+    /// a function to a contained value.
+    pub fn map<R>(self, f: impl FnOnce(T) -> R) -> AuthSuccess<R> {
+        AuthSuccess {
+            reported_auth_ok: self.reported_auth_ok,
+            value: f(self.value),
+        }
+    }
+}
+
 /// Info for establishing a connection to a compute node.
 /// This is what we get after auth succeeded, but not before!
+#[derive(Clone)]
 pub struct NodeInfo {
     /// Compute node connection params.
+    /// It's sad that we have to clone this, but this will improve
+    /// once we migrate to a bespoke connection logic.
     pub config: compute::ConnCfg,
+
     /// Labels for proxy's metrics.
-    pub aux: MetricsAuxInfo,
+    pub aux: Arc<MetricsAuxInfo>,
+}
+
+pub type NodeInfoCache = TimedLru<Arc<str>, NodeInfo>;
+pub type CachedNodeInfo = timed_lru::Cached<&'static NodeInfoCache>;
+
+/// Various caches for [`console`].
+pub struct ApiCaches {
+    /// Cache for the `wake_compute` API method.
+    pub node_info: NodeInfoCache,
 }
```

Comment:
Do you have plans for caching other API calls? I would suggest removing ApiCaches, and just use NodeInfoCache directly. Until we actually have other kinds of caches, this is an unnecessary complication, and just makes the code harder to understand.

If we introduce another kind of cache in the future, I'm not sure it would be an "API cache" anyway, and it might not be needed in the same places as NodeInfoCache.

Reply:
> Until we actually have other kinds of caches, this is an unnecessary complication, and just makes the code harder to understand.

That's a strange claim, because the name and the purpose are self-explanatory. Besides, there'll be one more cache (creds cache) per above, so I'm going to leave it as is.